datatypes/src/raster/data_type.rs

```diff
@@ -39,8 +40,8 @@ pub trait Pixel:
     + FromPrimitive<Self>
     + StaticRasterDataType
     + RgbaTransmutable
-    + TypedRasterConversion<Dim<[usize; 2]>>
-    + TypedRasterConversion<Dim<[usize; 3]>>
+    + TypedRasterConversion<Dim2D>
```

Comment:
You introduced 1D, what about 1D here?

Reply:
yes well it was not here before. i hope to find a way to remove the dimension dependency of the conversion trait.

Reply:
i think we can add it when the 1D thing gets used somewhere.